### ISSUE TITLE
Cannot use redis in CLI command due to missing igbinary loading

### ISSUE THREAD

**Issue by lbayerl** [cloudfoundry/php-buildpack#282]
What version of Cloud Foundry and CF CLI are you using?
**cf version 6.38.0+7ddf0aadd.2018-08-07**

What version of the buildpack you are using?
**4.3.64**

If you were attempting to accomplish a task, what was it you were attempting to do?
using "predis/predis": "~1.1"  in cli environment

What did you expect to happen?
I have a CLI command which clears the cache (incl Redis) which doesn't work.

What was the actual behavior?
_PHP Warning:  Cannot load module 'redis' because required module 'igbinary' is not loaded in Unknown on line 0_
This was working up until last version of buildpack used (4.3.61).

Don't know whether this is related to https://github.com/cloudfoundry/php-buildpack/issues/281 or maybe to https://github.com/cloudfoundry/php-buildpack/issues/273

Please confirm where necessary:
* [X] I have included a log output
* [X] My log includes an error message
* [ ] I have included steps for reproduction

**Comment by dmikusa-pivotal** [comment]
Make sure that you are sourcing the `.profile.d/*` scripts when you `cf ssh` into the container. These contain environment variables so that PHP is found on the path and also so that it can load extensions.

You can run `cf ssh "<app>" -t -c "/tmp/lifecycle/launcher /home/vcap/app bash ''"` and this will automatically source them for you.

The PR for #273 has been merged so I don't think that would be an issue. #281 could be it, but I'd expect errors about loading Hiredis not igbinary. If the above doesn't help, make sure you include more output like exact command being run & full output from that.

**Comment by lbayerl** [comment]
Thanks, I am pretty sure, that I correctly sourced profile.d because I am usually not able to call my "bin/console cache:clear" (Symfony command based on standard PHP CLI) without it. But any other command which is part of my cache clearing works without any error.

**Comment by Ben16** [comment]
Hi @lbayerl,

It seems like your issue might indeed be related to https://github.com/cloudfoundry/php-buildpack/issues/281. We've pushed a fix to that issue on the develop branch. Could you try using that and see if it fixes your issue?

In the event that this doesn't fix your issue, could you provide us with a sample fixture or other steps to reproduce it?

**Comment by lbayerl** [comment]
Using 
```
buildpacks:
  - https://github.com/cloudfoundry/php-buildpack.git#develop"
```
results in

> Updating app with these attributes...
>  name:                MYNAME
>  buildpacks:
>-   https://github.com/cloudfoundry/php-buildpack.git
>+   https://github.com/cloudfoundry/php-buildpack.git#develop

but installs _-------> Buildpack version 4.3.64_

Did I test the develop path correctly?

Following this approach this is what I do:

1. cf ssh MYNAME
2. HOME=$HOME/app source app/.profile.d/finalize_bp_env_vars.sh
3. cd app
4. bin/console cache:clear --env=prod

Step 4 is executing a PHP CLI command - but I just found out that simply calling ```php``` from app directory is resulting in

> PHP Warning:  Cannot load module 'redis' because required module 'igbinary' is not loaded in Unknown on line 0

For your info, this is my content regarding extensions in my app's php.ini.d folder:

```
zend_extension = ioncube.so
;ioncube has to be first line here!
zend_extension = opcache.so

; commented extensions are required but preconfigured by PHP buildpack already
;extension = bz2.so
;extension = zlib.so
;extension = curl.so
;extension = mcrypt.so
;extension = openssl.so
;extension = mbstring.so
;extension = pdo.so
extension = sockets.so
extension = pdo_mysql.so
extension = redis.so
extension = apcu.so
```

**Comment by dmikusa-pivotal** [comment]
Can you try adding `extension = igbinary.so`?  In #273, we enabled igbinary support for the redis extension and this had the side-effect of requiring both extensions to be enabled, whether you're using igbinary w/redis or not.

**Comment by lbayerl** [comment]
Thanks @dmikusa-pivotal , that seemed to be the issue. Pretty tricky that "out of a sudden" a new extension is required. Thanks again.

**Comment by dmikusa-pivotal** [comment]
Yea, sorry about that. That's my fault. When I fixed #273 I didn't expect it to require igbinary unconditionally. I was thinking it would only be necessary when the redis extension used the igbinary support. Didn't find this out until after I had made the changes.

I think we could perhaps allow the buildpack to help correct this. If it sees redis.so enabled, but not igbinary.so then we could automatically enable igbinary.so.
